src/app/(explore)/sources/[...slug]/page.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+import { ContentTreeArray } from "@/utils/data";
+import allSources from "@/public/sources-data.json";
+import { allTranscripts } from "contentlayer/generated";
+import BreadCrumbs from "@/components/common/BreadCrumbs";
+import { ArrowLinkRight } from "@bitcoin-dev-project/bdp-ui/icons";
+import { ContentTree, extractDirectoryData, filterOutIndexes } from "@/utils";
+import TranscriptDetailsCard from "@/components/common/TranscriptDetailsCard";
+
+export function generateStaticParams() {
+  return allTranscripts.reduce((acc, { _raw: { flattenedPath }, slugAsParams }) => {
```

Comment:
generateStaticParams function is not very readable.

Also a couple of observations:
- slugAsParams is an array of strings and not a string i.e `['adopting-bitcoin', '_index']` it was incorrectly typed from contentlayer it seems.

- we actually do not want to get rid of paths with `_index`, rather convert them to base route. i.e from `['adopting-bitcoin', '_index']` to `['adopting-bitcoin']`.

- continuing from second point if we do this conversion we have to account for other non English routes. i.e  ['adopting-bitcoin', '_index.es']` we would like to skip converting this because we will then have multiple base routes i.e multiple `['adopting-bitcoin']` routes.?

- the routes returned from `generateStaticParams` should be typed as `{ slug: string[] }[]` instead of `{ slug: string }[]`.

- `transcript._raw.flattenedPath.split("/")` and `slugAsParams` are essentially the same.

- force 404 for paths not generated from `getStaticParams` [see docs](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamicparams). We do not want to generate them server-side. This segues into the invocation timeout errors as we then build the page on navigation. If it hasn't been statically generated then it should be a 404.

Skipping base route on `generateStaticParams` means when one navigates to `/adopting-bitcoin` for instance it invokes the serverless function to create that page (hence the timeout or latency to render). Because `['adopting-bitcoin']` as a slug doesn't exist and wasn't statically generated.

Reply:
I did a small refactor with comments that fixes the issue described above:

```ts
// forces 404 for paths not generated from `generateStaticParams` function.
export const dynamicParams = false

export function generateStaticParams() {
  const all = allTranscripts.reduce((accumulator, transcript) => {

    const params = transcript._raw.flattenedPath.split("/");
    // transcript.slugAsParams is incorrectly typed as `string` this manually casts it to `string[]`
    const slugAsParams = transcript.slugAsParams as unknown as string[];

    const is_non_english_dir_or_transcript = /\w+\.[a-z]{2}\b/.test(params[params.length - 1])

    const lastRouteIsIndex = params[params.length - 1].includes("index");

    if (!lastRouteIsIndex) {
      accumulator.push({ slug: slugAsParams });
    }
    // converts slug ['adopting-bitcoin', '_index'] to ['adopting-bitcoin']
    // skips non english dir e.g ['adopting-bitcoin', '_index.es'] is skipped
    else if (!is_non_english_dir_or_transcript) {
      accumulator.push({ slug: slugAsParams.slice(0, -1) });
    }

    return accumulator;
    // { slug: string }[] should be { slug: string[] }[], due to nested slug, i.e [...slug] and not [slug]
  }, [] as { slug: string[] }[]);
  return all;
}
```

Reply:
cc: @IgboPharaoh

Reply:
Great review, I'll work on the update

Reply:
changes have been pushed

Reply:
Nice  👌